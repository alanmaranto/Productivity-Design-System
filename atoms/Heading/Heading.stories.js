import { Heading, styles, options } from ".";
import {
  getOptionsArgTypes,
  getTemplate,
  getListTemplate,
} from "../../helpers/storybook";

export default {
  title: "Atoms/Heading",
  component: Heading,
  args: {
    children: "Consent doubt TuckBorleje challeng destroying",
  },
  argTypes: {
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.sizes),
    weight: getOptionsArgTypes(options.weights),
    children: { control: "text" },
  },
};

const Template = getTemplate(Heading, styles);
const ListTemplate = getListTemplate(Heading, styles);

export const Default = Template.bind({});

export const Colors = ListTemplate.bind({});
Colors.args = {
  items: options.colors.map((color) => ({ color })),
};

export const Sizes = ListTemplate.bind({});
Sizes.args = {
  items: options.sizes.map((size) => ({ size })),
};

export const Weights = ListTemplate.bind({});
Weights.args = { items: options.weights.map((weight) => ({ weight })) };
