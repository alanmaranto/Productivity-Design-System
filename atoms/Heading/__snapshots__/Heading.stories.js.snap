// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Heading Colors 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <header
    className="color-base size-md weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
  <header
    className="color-primary size-md weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
  <header
    className="color-inverted size-md weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
</div>
`;

exports[`Storyshots Atoms/Heading Default 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <header
    className="color-base size-md weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
</div>
`;

exports[`Storyshots Atoms/Heading Sizes 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <header
    className="color-base size-xs weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
  <header
    className="color-base size-sm weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
  <header
    className="color-base size-md weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
  <header
    className="color-base size-lg weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
  <header
    className="color-base size-xl weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
  <header
    className="color-base size-2xl weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
</div>
`;

exports[`Storyshots Atoms/Heading Weights 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <header
    className="color-base size-md weight-normal heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
  <header
    className="color-base size-md weight-bold heading"
  >
    Consent doubt TuckBorleje challeng destroying
  </header>
</div>
`;
